store/src/java/com/zimbra/cs/service/mail/GetContacts.java

```diff
@@ -74,48 +79,50 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
                 throw ServiceException.FAILURE("Got remote folderId: " + folderIdStr + " but did not proxy", null);
         }
 
-        SortBy sort = SortBy.of(request.getAttribute(MailConstants.A_SORTBY, null));
+        SortBy sort = SortBy.of(req.getSortBy());
         if (sort == null) {
             sort = SortBy.NONE;
         }
         ArrayList<String> attrs = null;
         ArrayList<String> memberAttrs = null;
         ArrayList<ItemId> ids = null;
 
-        for (Element e : request.listElements()) {
-            if (e.getName().equals(MailConstants.E_ATTRIBUTE)) {
-                String name = e.getAttribute(MailConstants.A_ATTRIBUTE_NAME);
-                if (attrs == null)
-                    attrs = new ArrayList<String>();
-                attrs.add(name);
-            } else if (e.getName().equals(MailConstants.E_CONTACT_GROUP_MEMBER_ATTRIBUTE)) {
-                String name = e.getAttribute(MailConstants.A_ATTRIBUTE_NAME);
-                if (memberAttrs == null)
-                    memberAttrs = new ArrayList<String>();
-                memberAttrs.add(name);
-            } else if (e.getName().equals(MailConstants.E_CONTACT)) {
-                String idStr = e.getAttribute(MailConstants.A_ID);
-                String targets[] = idStr.split(",");
-                for (String target : targets) {
-                    ItemId iid = new ItemId(target, zsc);
-                    if (ids == null)
-                        ids = new ArrayList<ItemId>();
-                    ids.add(iid);
-                }
+        //MailConstants.A_ATTRIBUTE_NAME
+        List<AttributeName> reqAttrs = req.getAttributes();
+        if(reqAttrs != null && reqAttrs.size() > 0) {
+            attrs = new ArrayList<String>();
+            for(AttributeName attrName : reqAttrs) {
+                attrs.add(attrName.getName());
+            }
+        }
 
-                // remove it from the request, so we can re-use the request for proxying below
-                e.detach();
+        //MailConstants.E_CONTACT_GROUP_MEMBER_ATTRIBUTE
+        reqAttrs = req.getMemberAttributes();
+        if(reqAttrs != null && reqAttrs.size() > 0) {
+            memberAttrs = new ArrayList<String>();
+            for(AttributeName attrName : reqAttrs) {
+                memberAttrs.add(attrName.getName());
+            }
+        }
+
+        //MailConstants.E_CONTACT
+        List<Id> contactIds = req.getContacts();
+        if(contactIds != null && contactIds.size() > 0) {
```

Comment:
you are right. soap.txt:

<GetContactsRequest [sortBy="nameAsc|nameDesc"] [sync="1"] [l="{folder-id}"] [derefGroupMember="1"] 
    [returnHiddenAttrs="1"] [maxMembers="{max-members}"] [returnCertInfo="1"]>
  <a n="..."/>*
  <ma n="..."/>*
  <cn id="{contact-id}[,{contact-id}...]"/>*
</GetContactsRequest>